homepairs/homepairsUI/tests/redux/property-list/PropertyListReducer.test.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+    FetchPropertiesAction,
+    FetchPropertyAction,
+    AddPropertyAction,
+    RemovePropertyAction,
+    Property,
+    PropertyListState,
+    UpdatePropertyAction,
+    SetSelectedPropertyAction,
+} from 'homepairs-types';
+import { PropertyListActions } from 'homepairs-redux-actions';
+import { properties, initialState } from '../../../src/state/property-list/reducer';
+
+const { PROPERTY_LIST_ACTION_TYPES } = PropertyListActions;
+
+const PMProperties: Property[] = [
+    {
+        city: 'San Luis Obispo',
+        tenants: 5,
+        bathrooms: 2,
+        bedrooms: 3,
+        state: 'CA',
+        streetAddress: '200 N. Santa Rosa',
+    },
+    {
+        city: 'Glenndale',
+        tenants: 5,
+        bathrooms: 2,
+        bedrooms: 3,
+        state: 'CA',
+        streetAddress: '200 N. Santa Fey',
+    },
+    {
+        city: 'Dublin',
+        tenants: 5,
+        bathrooms: 2,
+        bedrooms: 3,
+        state: 'CA',
+        streetAddress: '200 New Fields Lane',
+    },
+];
+
+const testPropertyListStateWithoutIndex: PropertyListState = {
+    selectedPropertyIndex: undefined,
+    properties: PMProperties,
+};
+
+const testPropertyListStateWithIndex: PropertyListState = {
+    selectedPropertyIndex: 1,
+    properties: PMProperties,
+};
+
+describe('PropertyList Reducer Test', () => {
+
+    describe('Test action with Invalid Property type', () => {
+        it('Undefined State', () => {
+            const testAction: FetchPropertiesAction = {
+                type: 'INVALID_STATE',
+                properties: [],
+            };
+            const updatedProperties = properties(undefined, testAction);
+            expect(updatedProperties).toStrictEqual(initialState);
+        });
+
+        it('Defined State', () => {
+            const testAction: FetchPropertiesAction = {
+                type: 'INVALID_STATE',
+                properties: [],
+            };
+            const updatedProperties = properties(testPropertyListStateWithoutIndex, testAction);
+            expect(updatedProperties).toStrictEqual(testPropertyListStateWithoutIndex);
+        });
+    });
+
+    describe('Test action with ADD_PROPERTY type', () => {
+        const testProperty: Property = {
+            city: 'San Luis Obispo',
+            tenants: 5,
+            bathrooms: 2,
+            bedrooms: 3,
+            state: 'CA',
+            streetAddress: '200 N. Santa Rosa',
+        };
+        const testAction: AddPropertyAction = {
+            type: PROPERTY_LIST_ACTION_TYPES.ADD_PROPERTY,
+            userData: testProperty,
+        };
+        it('Default State', () => {
```

Comment:
these tests could be a tad more context (or im just bad with reducers) with what the default state is, since it'd offer explanation why remove/update don't have a default state. this certainly isn't required though